### ISSUE TITLE
[MediaLibrary] [IOS] Exaggerated Permissions for createAssetAsync

### ISSUE THREAD

**Issue by erandagan** [expo/expo#5089]
## TLDR
iOS 11+ added a write only permission to the camera roll. `MediaLibrary` needlessly enforces read+write permissions on methods that perform write only operations, leading to exaggerated permissions being requested.

### Environment
- Targeting iOS (10.3+) & Android (API 21+)
- Not using the Expo SDK (Unimodules setup)

```
react-native-unimodules@0.5.0
expo-media-library@6.0.0
```

### Steps to Reproduce

1. Add `NSPhotoLibraryAddUsageDescription` (**without adding** `NSPhotoLibraryUsageDescription`) to `Info.plist`
1. Download an image to the local filesystem (`localUri`)
1. Call `MediaLibrary.createAssetAsync(localUri)`

### Expected Behavior

Asset creation should succeed on iOS 11+.

### Actual Behavior
An error stating `CAMERA_ROLL permission is required to do this operation` will be thrown from [this line.](https://github.com/expo/expo/blob/40387ef9d42f8fc8df808dc28335e515056b1684/packages/expo-media-library/ios/EXMediaLibrary/EXMediaLibrary.m#L868)

### The Cause
I apologise for not supplying a demo, instead I'll describe what's causing this issue and how to fix it.

iOS 11.0 added a new permission key called `NSPhotoLibraryAddUsageDescription`, which provides **write only** access to the user's photos. This allows apps to save photos to the device, without being able to read the user's photos.

However, [this line](https://github.com/expo/expo/blob/master/packages/expo-media-library/ios/EXMediaLibrary/EXMediaLibrary.m#L868), originally called from [here](https://github.com/expo/expo/blob/40387ef9d42f8fc8df808dc28335e515056b1684/packages/expo-media-library/ios/EXMediaLibrary/EXMediaLibrary.m#L84) checks for the `CAMERA_ROLL` permission, which Expo defines as **both read & write**.

This check causes the **read** permission to be needlessly enforced for **write** operations on devices running iOS 11+ (to date, over 90% of the iOS devices in the world)

### Fixing this issue
- It's possible to split `_checkPermissions` to `_checkReadPermissions` and `_checkWritePermissions` and call these as required by the MediaLibrary methods. On devices running iOS < 11.0 (<10%), checkReadPermissions can fall back to enforce the `NSPhotoLibraryUsageDescription` key.
